PlanetWorker/src/PlanetWorker.cs

```diff
@@ -56,19 +67,140 @@ static int Main(string[] arguments)
               Environment.Exit(ErrorExitStatus);
             }
           });
+          
+          dispatcher.OnAuthorityChange<PlanetInfo>(op =>
+          {
+            var logMessage = String.Format("Changing Authority for entityId {0}", op.EntityId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+
+            ViewEntity entity;
+            if (EntityView.TryGetValue(op.EntityId, out entity))
+            {
+              if(op.Authority == Authority.Authoritative)
+              {
+                  entity.hasAuthority = true;
+              }
+              else if (op.Authority == Authority.NotAuthoritative)
+              {
+                  entity.hasAuthority = false;
+              }
+              else if (op.Authority == Authority.AuthorityLossImminent)
+              {
+                  entity.hasAuthority = false;
+              }
+            }
+          });
+          
+          dispatcher.OnAddComponent<PlanetInfo>(op =>
+          {
+            var logMessage = String.Format("Adding PlanetInfo Component for entityId {0}", op.EntityId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+
+            ViewEntity entity;
+            if (EntityView.TryGetValue(op.EntityId, out entity))
+            {
+                entity.entity.Add<PlanetInfo>(op.Data);
+            }
+            else
+            {
+                ViewEntity newEntity = new ViewEntity();
+                EntityView.Add(op.EntityId, newEntity);
+                newEntity.entity.Add<PlanetInfo>(op.Data);
+            }
+          });
+          
+          dispatcher.OnComponentUpdate<PlanetInfo>(op=>
+          {
+            var logMessage = String.Format("Updating PlanetInfo Component for entityId {0}", op.EntityId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+
+            ViewEntity entity;
+            if (EntityView.TryGetValue(op.EntityId, out entity))
+            {
+                var update = op.Update.Get();
+                entity.entity.Update<PlanetInfo>(update);
+            }
+          });
+          
+          dispatcher.OnAddEntity(op =>
+          {
+            var logMessage = String.Format("Adding entityId {0}", op.EntityId);
+            connection.SendLogMessage(LogLevel.Info, LoggerName, logMessage);
+
+            // AddEntity will always be followed by OnAddComponent
+            ViewEntity newEntity = new ViewEntity();
+            newEntity.hasAuthority = true;
```

Comment:
This needs to be false by default, then you'll receive an authority received op afterwards to fix it